pinot-broker/src/main/java/com/linkedin/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -0,0 +1,376 @@
+      LOGGER.info("Caught exception while compiling request {}: {}, {}", requestId, query, e.getMessage());
+      _brokerMetrics.addMeteredGlobalValue(BrokerMeter.REQUEST_COMPILATION_EXCEPTIONS, 1);
+      return new BrokerResponseNative(QueryException.getException(QueryException.PQL_PARSING_ERROR, e));
+    }
+    String tableName = brokerRequest.getQuerySource().getTableName();
+    String rawTableName = TableNameBuilder.extractRawTableName(tableName);
+    _brokerMetrics.addPhaseTiming(rawTableName, BrokerQueryPhase.REQUEST_COMPILATION,
+        System.nanoTime() - compilationStartTimeNs);
+    _brokerMetrics.addMeteredTableValue(rawTableName, BrokerMeter.QUERIES, 1);
+
+    // Check table access
+    long authorizationStartTimeNs = System.nanoTime();
+    boolean hasAccess = _accessControlFactory.create().hasAccess(requesterIdentity, brokerRequest);
+    if (!hasAccess) {
+      _brokerMetrics.addMeteredTableValue(brokerRequest.getQuerySource().getTableName(),
+          BrokerMeter.REQUEST_DROPPED_DUE_TO_ACCESS_ERROR, 1);
+      return new BrokerResponseNative(QueryException.ACCESS_DENIED_ERROR);
+    }
+    _brokerMetrics.addPhaseTiming(rawTableName, BrokerQueryPhase.AUTHORIZATION,
+        System.nanoTime() - authorizationStartTimeNs);
+
+    // Get the tables hit by the request
+    String offlineTableName = null;
+    String realtimeTableName = null;
+    CommonConstants.Helix.TableType tableType = TableNameBuilder.getTableTypeFromTableName(tableName);
+    if (tableType == CommonConstants.Helix.TableType.OFFLINE) {
+      // Offline table
+      if (_routingTable.routingTableExists(tableName)) {
+        offlineTableName = tableName;
+      }
+    } else if (tableType == CommonConstants.Helix.TableType.REALTIME) {
+      // Realtime table
+      if (_routingTable.routingTableExists(tableName)) {
+        realtimeTableName = tableName;
+      }
+    } else {
+      // Hybrid table (check both OFFLINE and REALTIME)
+      String offlineTableNameToCheck = TableNameBuilder.OFFLINE.tableNameWithType(tableName);
+      if (_routingTable.routingTableExists(offlineTableNameToCheck)) {
+        offlineTableName = offlineTableNameToCheck;
+      }
+      String realtimeTableNameToCheck = TableNameBuilder.REALTIME.tableNameWithType(tableName);
+      if (_routingTable.routingTableExists(realtimeTableNameToCheck)) {
+        realtimeTableName = realtimeTableNameToCheck;
+      }
+    }
+    if ((offlineTableName == null) && (realtimeTableName == null)) {
+      // No table matches the request
+      LOGGER.info("No table matches the name: {}", tableName);
+      _brokerMetrics.addMeteredGlobalValue(BrokerMeter.RESOURCE_MISSING_EXCEPTIONS, 1);
+      return BrokerResponseNative.NO_TABLE_RESULT;
+    }
+
+    // Validate QPS quota
+    if (!_tableQueryQuotaManager.acquire(tableName)) {
+      String errorMessage = String.format("Request %d exceeds query quota for table: %s", requestId, tableName);
+      LOGGER.info(errorMessage);
```

Comment:
Should it be `LOGGER.error`?

Reply:
No, this is not an error in Pinot. It is a user error. So we should log it as INFO and return 4xx to user.